src/contracts/Limitation/Target/DestinationLocation.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Contracts\Core\Limitation\Target;
+
+use eZ\Publish\SPI\Limitation\Target;
+use eZ\Publish\SPI\Persistence\ValueObject;
+
+/**
+ * Location Limitation target.
+ *
+ * @property-read int $id
+ * @property-read \eZ\Publish\API\Repository\Values\Content\ContentInfo $targetContentInfo
+ */
+class DestinationLocation extends ValueObject implements Target
+{
+    /**
+     * @var int
+     */
+    protected $id;
+
+    /**
+     * @var \eZ\Publish\API\Repository\Values\Content\ContentInfo
+     */
+    protected $targetContentInfo;
+}
```

Comment:
1. `final`
2. Quite fresh - we're getting performance hits due to `ValueObject::__get` magic. We should avoid it. Refactor this please to:

    - use strict constructor
    - use strict getters for the properties
    - it still can extend `ValueObject` for the consistency's sake, but should not utilize its magic.
    - CS: `/** @var <type> */` instead of multiline PHPDoc for the _properties_.